### PR TITLE
React routing

### PR DESCRIPTION
**Pull request by shwetzpatil**
Routing uses `react-router-component`(Declarative routing).
Currently it routes signin, login, post input and posts. Navigation bar is handle by header component and other pages( signup,login,posts)are handle by Main component.

### CODE REVIEW

**Comment by shwetzpatil** [comment]
[React-routing](https://trello.com/c/4L1et5v9/42-react-routing)

File: src/main/js/users/signUp.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        return (
+            <section className='signup'>
+                <h1 className='sign_up_header'>
+                    Please enter your details
+                </h1>
+                <form>
+                    <input type="email" name="email" placeholder="example@hello.com"/><br />
+                    <input type="password" name="password" placeholder="XXXXXX"/><br />
+                    <input type="text" name="name" placeholder="Xy Zello"/><br />
+                    <button type="submit" name="sign_up">Sign up</button>
```
Comment (TinyGobby) [inline src/main/js/users/signUp.js]: Maybe we should have more typical placeholders for the input fields?

File: src/main/js/header.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    render() {
+        return (
+            <header>
+                <nav>
+                    <ul>
+                        <li><Link to='/'>Home</Link></li>
+                        <li><Link to='/post'>posts</Link></li>
+                        <li><Link to='/login'>logIn</Link></li>
+                        <li><Link to='/signup'>SignUp</Link></li>
```
Comment (TinyGobby) [inline src/main/js/header.js:22]: What was the reason for separate login and signup pages?

**Review by TinyGobby — APPROVED** [review]
Excellent, nicely done

File: src/main/js/header.js (hunk 2 above)
Comment (shwetzpatil) [reply src/main/js/header.js:22]: login and signup have different template and eventually it will link to separate button as well.

**Review by Drescoding — APPROVED** [review]
Ran through the changes with Shweta, it makes sense and works.

File: src/main/js/users/signUp.js (hunk 1 above)
Comment (shwetzpatil) [reply src/main/js/users/signUp.js]: Yep, I agree. It is sample code just to whether it works or not ? :D

**Comment by racldn** [comment]
I went through this yesterday when Shweta was still here. Was waiting for the rest of the group to see it before reviewing and committing. Looks like its been merged now. Well done Shweta

**Pull request merged**